pom.xml

```diff
@@ -345,24 +361,28 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>${maven-compiler-plugin.version}</version>
                 <configuration>
-                    <source>${compiler.version}</source>
+                    <source>
```

Comment:
For JDK17, as I understand we should switch to `--release` argument of `javac` instead of keep using `source` and `target`. This can give us better guarantee of backward-compatibility with such cross-compilation.

- https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-release.html
- https://stackoverflow.com/questions/43102787/what-is-the-release-flag-in-the-java-9-compiler/43103038#43103038

Reply:
Sure, changed to `<release>` configuration, in plugin tests it's still `<source>` and `<target>` as the plugin tests are still built and run in Java 8

Reply:
I reverted the `<release>` back to `<source>` and `<target>` as that won't compile in our case, and because the plugin tests are still run in Java 8 that's OK enough to ensure we don't use APIs from > Java 8

Reply:
> I reverted the `<release>` back to `<source>` and `<target>` as that won't compile in our case, and because the plugin tests are still run in Java 8 that's OK enough to ensure we don't use APIs from > Java 8

I see. The `sun.*` is a disaster...